nerfstudio/data/datamanagers/full_images_datamanager.py

```diff
@@ -0,0 +1,412 @@
+            K = camera.get_intrinsics_matrices().numpy()
+            if camera.distortion_params is None:
+                continue
+            distortion_params = camera.distortion_params.numpy()
+            image = data["image"].numpy()
+
+            if camera.camera_type.item() == CameraType.PERSPECTIVE.value:
+                distortion_params = np.array(
+                    [
+                        distortion_params[0],
+                        distortion_params[1],
+                        distortion_params[4],
+                        distortion_params[5],
+                        distortion_params[2],
+                        distortion_params[3],
+                        0,
+                        0,
+                    ]
+                )
+                newK, roi = cv2.getOptimalNewCameraMatrix(K, distortion_params, (image.shape[1], image.shape[0]), 0)
+                image = cv2.undistort(image, K, distortion_params, None, newK) # type: ignore
+                # crop the image and update the intrinsics accordingly
+                x, y, w, h = roi
+                image = image[y : y + h, x : x + w]
+                if "mask" in data:
+                    data["mask"] = data["mask"][y : y + h, x : x + w]
+                if "depth_image" in data:
+                    data["depth_image"] = data["depth_image"][y : y + h, x : x + w]
+                K = newK
+                # update the width, height
+                self.train_dataset.cameras.width[i] = w
+                self.train_dataset.cameras.height[i] = h
+
+            elif camera.camera_type.item() == CameraType.FISHEYE.value:
+                distortion_params = np.array(
+                    [distortion_params[0], distortion_params[1], distortion_params[2], distortion_params[3]]
+                )
+                newK = cv2.fisheye.estimateNewCameraMatrixForUndistortRectify(
+                    K, distortion_params, (image.shape[1], image.shape[0]), np.eye(3), balance=0
+                )
+                map1, map2 = cv2.fisheye.initUndistortRectifyMap(
+                    K, distortion_params, np.eye(3), newK, (image.shape[1], image.shape[0]), cv2.CV_32FC1
+                )
+                # and then remap:
+                image = cv2.remap(image, map1, map2, interpolation=cv2.INTER_LINEAR, borderMode=cv2.BORDER_CONSTANT)
+                K = newK
+            else:
+                raise NotImplementedError("Only perspective and fisheye cameras are supported")
+            data["image"] = torch.from_numpy(image)
+
+            if "mask" in data:
+                mask = data["mask"].numpy()
+                if camera.camera_type.item() == CameraType.PERSPECTIVE.value:
+                    mask = cv2.undistort(mask, K, distortion_params, None, None) # type: ignore
+                elif camera.camera_type.item() == CameraType.FISHEYE.value:
+                    mask = cv2.fisheye.undistortImage(mask, K, distortion_params, None, None)
```

Comment:
I think this is a mismatch from the images since for the images we compute the new image by using newK - I think either we should reuse the map generated for the images or pass in newK into this call. 
```suggestion
                    mask = cv2.fisheye.undistortImage(mask, K, distortion_params, None, None)
```

Reply:
Also notice we dont undistort the depth image in this case. Maybe it would be cleaner to move both of this under the main fisheye condition?

Reply:
This was fixed in #2540 !